### PR TITLE
[service.subtitles.napisy24pl] 2.0.1

### PR DESCRIPTION
**Pull request by CaTzil**
### Description

Complete rewrite of the code, now allows to search tv-shows more accurately and faster. 
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: service.subtitles.napisy24pl/resources/lib/NapisyUtils.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+
+        f = self.urlHandler.request(self.BASE_URL + "/download", query, referer=self.BASE_URL)
+
+        if f is None and self.login(True):
+            f = self.urlHandler.request(self.BASE_URL + "/download", query, referer=self.BASE_URL)
+
+        with open(zip_filename, "wb") as subFile:
+            subFile.write(f)
+        subFile.close()
+        xbmc.sleep(500)
```
Comment (Tadly) [inline service.subtitles.napisy24pl/resources/lib/NapisyUtils.py]: Please use something like `xbmc.Monitor().waitForAbort(0.5)` instead.
We discourage the usage of `xbmc.sleep`

File: service.subtitles.napisy24pl/service.py [hunk 2]
```diff
@@ -306,6 +89,7 @@ def get_params():
 
     return param
 
+
 params = get_params()
```
Comment (Tadly) [inline service.subtitles.napisy24pl/service.py]: I just noticed that the stuff from here on out isn't encapsulated in anything.
Personally I'd prefer seeing this within a:
```python
if __name__ == '__main':
    ...
```
but isn't required to be changed.

Comment (CaTzil) [inline service.subtitles.napisy24pl/service.py]: This is how amet recommended to write subtitle addon, look at the source code of opensubtitle that he wrote:
https://github.com/xbmc/repo-scripts/blob/gotham/service.subtitles.opensubtitles/service.py#L128

Comment (Tadly) [inline service.subtitles.napisy24pl/service.py]: Hm...
I don't see a reason for the source to **not** be encapsulated in either their own methods or the typical _python main function_ as shown above.

I just think it's a bad practice unless it is required.
But as I said, no need to change it ;)

**Comment by Tadly** [comment]
LGTM

**Pull request merged**